### PR TITLE
[fuchsia] Bump the Fuchsia target API level to 10

### PR DESCRIPTION
**Pull request by sebmarchand**
Flutter has started using F10 SDK, it should now target API level 10.

### CODE REVIEW

File: shell/platform/fuchsia/flutter/tests/integration/text-input/text-input-test.cc [hunk 1]
```diff
@@ -112,6 +112,9 @@ class TestResponseListenerServer
     return match;
   }
 
+  // fuchsia::ui::test::input::KeyboardInputListene override
```
Comment (akbiggs) [inline shell/platform/fuchsia/flutter/tests/integration/text-input/text-input-test.cc:115]: nit: KeyboardInputListener

Comment (sebmarchand) [reply shell/platform/fuchsia/flutter/tests/integration/text-input/text-input-test.cc]: Fixed.

**Comment by filmil** [comment]
May I ask: the currently released API level is 9, why is Flutter using level 10?

**Comment by sebmarchand** [comment]
@filmil  it's a little bit tricky, but the tl;dr is that while it's true that the current released API level is 9, the one that is in-development is API level 10 and there's a few things that make it unsafe right now to not use the latest API level exposed by an SDK (go/fuchsia-forward-compat-trailhead explains some of the issues)

**Comment by filmil** [comment]
OK, thank you for explaining. I wasn't aware that Flutter must live on Fuchsia top-of-tree. But if it must, then that's what it is.

**Review by lin-erik — APPROVED** [review]

**Pull request merged**